### ISSUE TITLE
sensecap-indicator-ha  build issue on MacOs

### ISSUE THREAD

**Issue by facero** [Love4yzp/sensecap-indicator-ha#1]
Hi all,

I’m also trying to build the sensecap-indicator-ha without success so far due to library include (or some of my environment ?)
I’m on Mac M1 with MacOS v14.4.1. I don’t use VScode. Installed CMake, etc via HomeBrew.
Also installed Xcode.
Here are the steps I’ve followed so far for IDF install and build :

```
git clone --recursive https://github.com/espressif/esp-idf.git
git checkout tags/v5.1.4 (also tried 5.2, 5.3 and 5.4)
```

Then :

```
./install.sh
./install.sh esp32s3
. ./export.sh
```
All worked without error

Then :

```
git clone https://github.com/Love4yzp/sensecap-indicator-ha
idf.py build
```
Then the error I get is :
```
/Users/facero/Documents/Perso/HomeAssistant-SenseCAP/esp32/esp-idf/components/heap/tlsf/tlsf.c:10:10: fatal error: tlsf.h: No such file or directory
   10 | #include "tlsf.h"
      |          ^~~~~~~~
compilation terminated.
[409/1277] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32s3/esp_memprot.c.obj
ninja: build stopped: subcommand failed.
Missing "tlsf.h" file name found in the following component(s): heap(/Users/facero/Documents/Perso/HomeAssistant-SenseCAP/esp32/esp-idf/components/heap/tlsf/include/tlsf.h). 
Maybe one of the components needs to add the missing header directory to INCLUDE_DIRS of idf_component_register call in CMakeLists.txt.
```
I checked and the tlsf.h is present in the mentionned directory so don’t understand why it cannot find it .

Any idea how to solve this ?

Otherwise if someone is willing to share a binary ready to flash, that would also be super helpful.
Thanks !

**Comment by facero** [comment]
Updated, this error seems to be resolved using  `esp-idf v5.1.0` but  `esp-idf v5.1.4`  fails.

Maybe this could be specified in the README file that the version is **absolutely** required to be `esp-idf v5.1.0`  installed using the command :
`git clone -b v5.1 --recursive https://github.com/espressif/esp-idf.git esp-idf-v5.1`

as specified in  :  

https://github.com/espressif/esp-idf/releases/tag/v5.1

**Comment by Love4yzp** [comment]
> Updated, this error seems to be resolved using `esp-idf v5.1.0` but `esp-idf v5.1.4` fails.
> 
> Maybe this could be specified in the README file that the version is **absolutely** required to be `esp-idf v5.1.0` installed using the command : `git clone -b v5.1 --recursive https://github.com/espressif/esp-idf.git esp-idf-v5.1`
> 
> as specified in :
> 
> https://github.com/espressif/esp-idf/releases/tag/v5.1

I got it. I test it in v5.1.0 v5.1.1 and v5.1.3;
if there is error when using esp-idf v5.1.4; I'll try to test that version.
thanks.
